src/gc/objecthandle.cpp

```diff
@@ -1820,7 +1820,10 @@ int GetCurrentThreadHomeHeapNumber()
     WRAPPER_NO_CONTRACT;
 
     if (g_theGCHeap == nullptr)
```

Comment:
I do not think there can be any scenarios where this is called without GC to be initialized. Can this null check be turned into assert?

Reply:
yes this should be an assert

---
In reply to: [237706859](https://github.com/dotnet/coreclr/pull/21150#discussion_r237706859) [](ancestors = 237706859)